todo.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+/*
+PROJECT: TODO APP
+A todo app that takes in command line arguments and uses them to manage a to-do list in a text file.
+
+REQUIREMENTS:
+Add Task (Index, Date, Due Date, Description)
+Remove Task by ID
+View Tasks
+
+Store tasks in a text file.
+*/
+func main() {
+	/* Open the file, defer close. (os.O_RDWR is opening the file for read and write,
+	O_CREATE creates if does not exist, '0666 gives read and write permissions) */
+	tasks, err := os.OpenFile("tasks.txt", os.O_CREATE|os.O_RDWR, 0666)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer tasks.Close()
+
+	// Use the command line arguments to decide what to do
+	action := os.Args[1]
+
+	if action == "add-task" {
+		CheckArgLength(4, os.Args)
+		taskToAdd := os.Args[2]
+		dueDate := os.Args[3]
+		newId := GenerateTaskID(tasks)
+		AddTask(tasks, newId, taskToAdd, dueDate)
+	} else if action == "remove" {
+		CheckArgLength(3, os.Args)
+		taskToRemove := os.Args[2]
+		RemoveTask(tasks, taskToRemove)
+	} else if action == "list" {
+		ViewTasks(tasks)
+	} else {
+		log.Fatal("Invalid action provided.")
+	}
+}
+
+// Gets the largest ID in the file and returns a new id incremented by one
+func GenerateTaskID(tasks *os.File) int {
+	biggestID := 0
+	line := ""
+	// Regex to match the first number in the line.
+	re := regexp.MustCompile(`^\d+`)
```

Comment:
Pet Peeve: I think using a regex here is probably overkill. There is definitely be a simpler way to grab the largest number used and just add 1 to it. Regex makes code notoriously opaque and may be more trouble than it's worth.

Try to remember: **_"use code as communication" - Robert Martin_**

If regex is the least complex way of doing what you want to do, and regex is the best to communicate that, then have at it, but i'm not certain that's the case here.